### ISSUE TITLE
Ps_Customtext in 1.7.2.4 does not use call Hook filterHtmlContent for it's HTML Field

### ISSUE THREAD

**Issue by elburgl69** [PrestaShop/PrestaShop#18520]
#### Ps_Customtext in 1.7.2.4 through 1.7.6.4 does not use call Hook filterHtmlContent for it's HTML Field

PS_Customtext call the hook  filterHtmlContent the text (HTML) field. Therefore customs modules that rely on this hook do not function on the content of this module.

#### Expected behavior

The hook is called.

#### Steps to Reproduce

1. Implement a module that uses the Hook filterHtmlContent to filter the HTML Content being processed.
2. Put the filter text of the module (installed above) in the HTML field of PS_Customtext. fe. {Product:1254}
3. Look at home page where PS_Customtext is shown and the text shows the filter text (fe. {Product:1254}) instead of the filtered result.

I do not know of modules in standard prestatshop install that uses the Hook filterHtmlContent, so duplication is not possible on a standard installation. I can supply you with a simple test module that filters the HTML if required.

However the module ps_customtext has in ps_customtext.php the following code to retrieve the data from the database.
```
public function getWidgetVariables($hookName = null, array $configuration = [])
    {
        $sql = 'SELECT * FROM `'._DB_PREFIX_.'info_lang`
            WHERE `id_lang` = '.(int)$this->context->language->id.' AND  `id_shop` = '.(int)$this->context->shop->id;

        return array(
            'cms_infos' => Db::getInstance()->getRow($sql),
        );
    }
```
This clearly retrieves the information directly from the database without using the objectModel, while the hookHtmlContent is called in  /src/Adapter/ObjectPresenter.php according to the list of Hooks.

When I retrieve the data using the objectmodel and get the data using the object presenter the HTML content is filtered.

#### Additional information

* PrestaShop version: 1.7.2.4 up to 1.7.6.4
* PHP version: PHP 7.1.33 and 7.3.16

Made a quick fix (no additional languages or shops) to Ps_Customtext::getWidgetVariables(), New function pasted below.

 ```
public function getWidgetVariables($hookName = null, array $configuration = [])
 {
        $ct = new CustomText(1, (int) $this->context->language->id, (int) $this->context->shop->id);
        $op = new ObjectPresenter();
        $pobj = $op->present($ct);
        $pobj['id_lang'] = $this->context->language->id;
        $pobj['id_shop'] = $this->context->shop->id;
        unset($pobj['id']);
        $ret = [
            'cms_infos' => $pobj,
        ];
        return $ret;
}
```

**Comment by marionf** [comment]
@PrestaShop/prestashop-core-developers Could you try to reproduce it ? Thanks

**Comment by elburgl69** [comment]
I updated the original message. Replicating the issue is dificult since I do not know of a standard Prestatshop Module that uses the hook filterHtmlContent. If required I can upload a sample module that transforms f.e. the string {msg:This an example!!} to **This is an example!!** (<b>This is an example!!</b>)

**Comment by elburgl69** [comment]
Attached an example module to process the tekst {msg:Text} in the ps_customtext HTML field to <b>Text</b>.
To reproduce the problem: 
1. Install the attached module boldtext.
2. In module ps_customtext enter the text {msg:This is a Test!}
3. Goto home page look at the ps_customtext section. You see the tekst {msg:This is a Test!}. It should be **This is a Test!**

[boldtext.zip](https://github.com/PrestaShop/PrestaShop/files/4483415/boldtext.zip)

**Comment by PierreRambaud** [comment]
Hi @elburgl69 
If you need this change in the ps_customtext module, feel free to open a pull request :)

**Comment by hibatallahAouadni** [comment]
Hello @elburgl69 ,

Since we had no news from you for more than 20 days, I'll close this ticket to avoid cluttering up the backlog.
Please note that you can always create a new one if further information pops up.

Thanks!

**Comment by elburgl69** [comment]
I am creating a pull request for it, please reopen.

**Comment by hibatallahAouadni** [comment]
@elburgl69 ok :wink:

**Comment by Progi1984** [comment]
Fixed by PrestaShop/ps_customtext#53
